engine/src/lib.rs

```diff
@@ -142,6 +143,25 @@ mod contract {
         io.return_output(state.owner_id.as_bytes());
     }
 
+    /// Set owner account id for this contract.
+    #[no_mangle]
+    pub extern "C" fn set_owner() {
+        let mut io = Runtime;
+        let mut state = state::get_state(&io).sdk_unwrap();
+        require_owner_only(&state, &io.predecessor_account_id());
+        let args: SetOwnerArgs = io.read_input_borsh().sdk_unwrap();
+        if state.owner_id == args.new_owner {
+            // Would be a no-op to set, do nothing and return false
+            io.return_output(b"false");
+        } else {
+            state.owner_id = args.new_owner;
+            state::set_state(&mut io, state).sdk_unwrap();
+            io.return_output(b"true");
+        }
+        // return true as bytes
+        io.return_output(b"true");
```

Comment:
Why? With this output, you will get `falsetrue` or `truetrue`.

Reply:
That, and we normally don't return on `set_` unless there is an error. So on false, return error. On true, return nothing. There doesn't need to be a return in case of true.

Reply:
@hskang9 `io.return_output(b"true");` is redundant.

Reply:
The output return in line 159 should have been already removed.

Reply:
I mean on line 161.

Reply:
> There doesn't need to be a return in case of true.
⬆️

Reply:
Remove 160 and 161 lines.